app/controllers/bike_share_app.rb

```diff
@@ -1,3 +1,131 @@
+require 'will_paginate'
+require 'will_paginate/active_record'
+
 class BikeShareApp < Sinatra::Base
+include WillPaginate::Sinatra::Helpers
+
+  get '/' do
+    erb :homepage
+  end
+
+  get '/stations' do
+    @stations = Station.all
+    erb :"stations/index"
+  end
+
+  get '/stations/new' do
+    erb :"/stations/new"
+  end
+
+  get '/stations-dashboard' do
+    erb :"/stations/dashboard"
+  end
+
+  get '/stations/:id' do
+    @station = Station.find(params[:id])
+    erb :"stations/show"
+  end
+
+  post '/stations' do
+    station = Station.write(params[:stations])
+    redirect "/stations/#{ station.id }"
+  end
+
+  get '/stations/:id/edit' do
+    @station = Station.find(params[:id])
+    erb :"stations/edit"
+  end
+
+  put '/stations/:id' do
+    station = Station.find(params[:id])
+    station.write_update(station, params[:stations])
```

Comment:
since `write_update` is an instance method, do we need to pass in the specific `station` since we already have access to it?  you can access all the attributes within a specific instance of `Station` within an instance method 👍